### ISSUE TITLE
Can't Ctrl+C stop during Execution stage

### ISSUE THREAD

**Issue by wmitsuda** [ledgerwatch/erigon#5094]
v2022.08.02

Can't do a clean shutdown if you Ctrl+C while it is running stage 6, had to force interrupt to exit it:

```
INFO[08-17|16:53:01.981] [6/16 Execution] Executed blocks         number=15356193 blk/s=8.6 tx/s=1567.0 Mgas/s=131.9 gasState=0.04 estimated duration=13m19.007s batch=41.7MB alloc=6.0GB sys=6.5GB
INFO[08-17|16:53:05.394] [p2p] GoodPeers                          eth66=7
INFO[08-17|16:53:05.757] [txpool] stat                            block=15354705 pending=4460 baseFee=30000 queued=30000 alloc=3.3GB sys=6.6GB
INFO[08-17|16:53:21.982] [6/16 Execution] Executed blocks         number=15356359 blk/s=8.3 tx/s=1654.6 Mgas/s=126.7 gasState=0.05 estimated duration=13m11.788s batch=46.2MB alloc=4.0GB sys=6.6GB
^CINFO[08-17|16:53:27.149] Got interrupt, shutting down...
INFO[08-17|16:53:27.149] Exiting...
INFO[08-17|16:53:27.149] RPC server shutting down
INFO[08-17|16:53:27.149] Exiting Engine...
INFO[08-17|16:53:27.149] RPC server shutting down
INFO[08-17|16:53:27.149] RPC server shutting down
WARN[08-17|16:53:27.149] Failed to serve http endpoint            err="http: Server closed"
INFO[08-17|16:53:27.149] Engine HTTP endpoint close               url=localhost:8551
WARN[08-17|16:53:27.149] Failed to serve http endpoint            err="http: Server closed"
INFO[08-17|16:53:27.149] HTTP endpoint closed                     url=localhost:8545
EROR[08-17|16:53:27.240] Staged Sync                              err="[6/16 Execution] batch commit: stopped"
INFO[08-17|16:53:28.131] [6/16 Execution] Blocks execution        from=15354705 to=15360652
EROR[08-17|16:53:28.131] Staged Sync                              err="[6/16 Execution] batch commit: stopped"
INFO[08-17|16:53:29.008] [6/16 Execution] Blocks execution        from=15354705 to=15360652
EROR[08-17|16:53:29.008] Staged Sync                              err="[6/16 Execution] batch commit: stopped"
INFO[08-17|16:53:29.882] [6/16 Execution] Blocks execution        from=15354705 to=15360652
```

(repeats forever)

**Comment by AskAlexSharov** [comment]
fixed by https://github.com/ledgerwatch/erigon/pull/5194

**Comment by revitteth** [comment]
Closing as fixed by #5194
